### PR TITLE
fix(domains-zone-edit): cname conflict text

### PR DESCRIPTION
**Pull request by ganeshkumar1989**
Close MBE-277

### Requirements

While creating a cname record, if there are conflicting records for the same sub-domain, the user should be advised to delete those records (with the list of records)

### Description of the Change

The error text has been modified, also listing the conflicting records.

### CODE REVIEW

File: client/app/domain/translations/Messages_fr_FR.xml [hunk 1]
```diff
@@ -519,7 +519,8 @@
    <translation id="domain_configuration_dns_entry_add_step3_alert" qtlid="195048">La modification sera immédiate dans la zone DNS, mais veuillez prendre en compte le temps de propagation (maximum 24h).</translation>
 
    <translation id="domain_configuration_dns_entry_add_step3_check_warning1" qtlid="215065">Attention, un ou plusieurs conflits peuvent survenir :</translation>
-   <translation id="domain_configuration_dns_entry_add_step3_check_warning2" qtlid="519831">Le sous-domaine est déjà utilisé par un autre enregistrement. Un enregistrement CNAME n'est pas autorisé à coexister avec d'autres champs sur le même sous-domaine.</translation>
+   <translation id="domain_configuration_dns_entry_add_step3_check_warning2">Le sous-domaine est déjà utilisé par un autre enregistrement. Un enregistrement CNAME n'est pas autorisé à coexister avec d'autres champs sur le même sous-domaine. Supprimez les enregistrements suivants pour le même sous-domaine, afin de pouvoir ajouter cet enregistrement:</translation>
+   <translation id="domain_configuration_dns_entry_add_step3_conflicting_record">{0} {1} {2}</translation>
```
Comment (antleblanc) [inline client/app/domain/translations/Messages_fr_FR.xml]: It doesn't make sense to translate this kind of sentence.

Prefer using ngBind on your view.

File: client/app/domain/translations/Messages_fr_FR.xml [hunk 2]
```diff
@@ -519,7 +519,8 @@
    <translation id="domain_configuration_dns_entry_add_step3_alert" qtlid="195048">La modification sera immédiate dans la zone DNS, mais veuillez prendre en compte le temps de propagation (maximum 24h).</translation>
 
    <translation id="domain_configuration_dns_entry_add_step3_check_warning1" qtlid="215065">Attention, un ou plusieurs conflits peuvent survenir :</translation>
-   <translation id="domain_configuration_dns_entry_add_step3_check_warning2" qtlid="519831">Le sous-domaine est déjà utilisé par un autre enregistrement. Un enregistrement CNAME n'est pas autorisé à coexister avec d'autres champs sur le même sous-domaine.</translation>
+   <translation id="domain_configuration_dns_entry_add_step3_check_warning2">Le sous-domaine est déjà utilisé par un autre enregistrement. Un enregistrement CNAME n'est pas autorisé à coexister avec d'autres champs sur le même sous-domaine. Supprimez les enregistrements suivants pour le même sous-domaine, afin de pouvoir ajouter cet enregistrement:</translation>
```
Comment (antleblanc) [inline client/app/domain/translations/Messages_fr_FR.xml]:
```suggestion
   <translation id="domain_configuration_dns_entry_add_step3_check_warning2">Le sous-domaine est déjà utilisé par un autre enregistrement. Un enregistrement CNAME n'est pas autorisé à coexister avec d'autres champs sur le même sous-domaine. Supprimez les enregistrements suivants pour le même sous-domaine, afin de pouvoir ajouter cet enregistrement :</translation>
```

File: client/app/domain/translations/Messages_fr_FR.xml (hunk 1 above)
Comment (ganeshkumar1989) [reply client/app/domain/translations/Messages_fr_FR.xml]: This is done :)

File: client/app/domain/translations/Messages_fr_FR.xml (hunk 2 above)
Comment (ganeshkumar1989) [reply client/app/domain/translations/Messages_fr_FR.xml]: This is fixed

File: client/app/domain/zone/record/domain-zone-record.controller.js [hunk 3]
```diff
@@ -93,8 +93,9 @@ angular.module('App').controller(
         subDomain: punycode.toASCII(this.model.subDomainToDisplay || ''),
         target: this.model.target.value,
       })
-        .then((isOk) => {
-          this.recordConflicts = isOk;
+        .then((result) => {
```
Comment (antleblanc) [inline client/app/domain/zone/record/domain-zone-record.controller.js]: You can use destructuring:
```suggestion
        .then(({ recordCanBeAdded, conflictingRecords }) => {
```

In that way you are able to do::
```js
this.recordConflicts = recordCanBeAdded;
this.conflictingRecords = conflictingRecords;
```

File: client/app/domain/domain.service.js [hunk 4]
```diff
@@ -597,7 +600,7 @@ angular.module('services').service(
             'CNAME',
             subDomain,
             entry.excludeId,
-          ).then(haveNoBrothers => haveNoBrothers);
+          ).then(haveNoBrothers => ({ recordCanBeAdded: haveNoBrothers }));
```
Comment (FredericEspiau) [inline client/app/domain/domain.service.js]: Could you please rename the whole `brother` thing please ? It's neither correct English neither a technical term.

You could for example they `subDomainsWithSameFieldTypes`

File: client/app/domain/zone/record/domain-zone-record.controller.js (hunk 3 above)
Comment (ganeshkumar1989) [reply client/app/domain/zone/record/domain-zone-record.controller.js]: Thanks for the suggestion. This is done :)

File: client/app/domain/domain.service.js (hunk 4 above)
Comment (ganeshkumar1989) [reply client/app/domain/domain.service.js]: All brother things have been removed :)

File: client/app/domain/domain.service.js [hunk 5]
```diff
@@ -513,11 +513,11 @@ angular.module('services').service(
      * @param {string} subDomain
      * @param {string} excludeId
      */
-    checkIfRecordHaveNoBrothers(serviceName, fieldType, subDomain, excludeId) {
+    subdomainExistsForFieldType(serviceName, fieldType, subDomain, excludeId) {
```
Comment (FredericEspiau) [inline client/app/domain/domain.service.js]:
```suggestion
    getExistingSubdomains(serviceName, fieldType, subDomain, excludeId) {
```

😄

File: client/app/domain/domain.service.js [hunk 6]
```diff
@@ -587,17 +587,20 @@ angular.module('services').service(
...
+            return {
+              recordCanBeAdded: !existingSubDomain.length,
+              conflictingRecords: existingSubDomain,
+            };
           });
         // Rule for Other Record types:
         // Can not have a CNAME of this subDomain
         default:
-          return this.checkIfRecordHaveNoBrothers(
+          return this.subdomainExistsForFieldType(
```
Comment (FredericEspiau) [inline client/app/domain/domain.service.js]:
```suggestion
          return this.getExistingSubdomains(
```

File: client/app/domain/domain.service.js (hunk 5 above)
Comment (ganeshkumar1989) [reply client/app/domain/domain.service.js]: Renamed :)

File: client/app/domain/domain.service.js (hunk 6 above)
Comment (ganeshkumar1989) [reply client/app/domain/domain.service.js]: Done

File: client/app/domain/domain.service.js (hunk 5 above)
Comment (FredericEspiau) [reply client/app/domain/domain.service.js]: Awesome

File: client/app/domain/zone/record/resume.html [hunk 7]
```diff
@@ -19,14 +19,20 @@
...
 
-    <div class="alert alert-danger mt-4" role="alert"
-         data-ng-if="!ctrl.recordConflicts">
-        <div data-ng-if="ctrl.model.fieldType === 'CNAME'"
-             data-translate="domain_configuration_dns_entry_add_step3_check_warning2">
-        </div>
-        <div data-ng-if="ctrl.model.fieldType !== 'CNAME'"
-             data-translate="domain_configuration_dns_entry_add_step3_check_warning3">
+    <div data-ng-if="!ctrl.recordConflicts">
+        <div class="alert alert-danger mt-4" role="alert">
```
Comment (antleblanc) [inline client/app/domain/zone/record/resume.html]: Could be turned into a `<oui-message>`?

Comment (ganeshkumar1989) [reply client/app/domain/zone/record/resume.html]: This is done :)

**Pull request merged**